modules/smaatoBidAdapter.js

```diff
@@ -0,0 +1,271 @@
+import * as utils from '../src/utils.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { config } from '../src/config.js';
+import { BANNER, VIDEO } from '../src/mediaTypes.js';
+
+const BIDDER_CODE = 'smaato';
+const SMAATO_ENDPOINT = 'https://prebid.ad.smaato.net/oapi/prebid';
+const CLIENT = 'prebid_js_$prebid.version$_1.0'
+
+/**
+* Transform BidRequest to OpenRTB-formatted BidRequest Object
+* @param {Array<BidRequest>} validBidRequests
+* @param {any} bidderRequest
+* @returns {string}
+*/
+const buildOpenRtbBidRequestPayload = (validBidRequests, bidderRequest) => {
+  /**
+   * Turn incoming prebid sizes into openRtb format mapping.
+   * @param {*} sizes in format [[10, 10], [20, 20]]
+   * @returns array of openRtb format mappings [{w: 10, h: 10}, {w: 20, h: 20}]
+   */
+  const parseSizes = (sizes) => {
+    return sizes.map((size) => {
+      return {w: size[0], h: size[1]};
+    })
+  }
+
+  const imp = validBidRequests.map(br => {
+    const bannerMediaType = utils.deepAccess(br, 'mediaTypes.banner');
+    const videoMediaType = utils.deepAccess(br, 'mediaTypes.video');
+    let result = {
+      id: br.bidId,
+      tagid: utils.deepAccess(br, 'params.adspaceId')
+    }
+
+    if (bannerMediaType) {
+      const sizes = parseSizes(utils.getAdUnitSizes(br));
+      result.banner = {
+        w: sizes[0].w,
+        h: sizes[0].h,
+        format: sizes
```

Comment:
This was done on purpose, since Smaato's bidder endpoint is currently transitioning from OpenRTB 2.3 to OpenRTB 2.5 (format is supported from 2.5 on) and I needed to keep compatible with both versions. This will probably change once the transition on the backend side completed.